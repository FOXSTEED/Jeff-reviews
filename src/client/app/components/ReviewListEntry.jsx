/* eslint-disable react/prop-types */

import React from 'react';
import styles from './styling/ReviewListEntry.css';
import ReviewBubbles from './ReviewBubbles';

const ReviewListEntry = props => (
  <div className={styles.main}>
    <div className={styles.user}>
      <img src={props.review.userImage} className={styles.picture}/>
      <p className={styles.username}>{props.review.userName}</p>
      <p className={styles.location}>{props.review.userLocation}</p>
      <span className={styles.belowUser}>{props.review.userReviews} <i class="far fa-edit"></i></span>
      <span className={styles.belowUser}>{props.review.userThumbs} <i class="fas fa-thumbs-up"></i></span>
    </div>
    <div>
      <div className={styles.header}>
        <span className={styles.bubbles}><ReviewBubbles rating={props.review.rating}/></span>
        <span className={styles.date}>{`Reviewed ${props.review.howRecent}`}</span>
      </div>
      <div className={styles.comment}>
        <p className={styles.headline}>{props.review.reviewHeadline}</p>
        <p className={styles.review}>{props.review.comment}</p>
      </div>
      <div>
        <a href='#' className={styles.ask}>{`Ask ${props.review.userName} about ${props.review.location}`}</a>
        <div>
          <button className={styles.button}><span className={styles.thumb}><i class="fas fa-thumbs-up"></i></span>{`Thank ${props.review.userName}`}</button>
          <span className={styles.flag}><i class="fas fa-flag"></i></span>
        </div>
        <p className={styles.bottomnote}><em>This review is the subjective opinion of a Nomad member and not of Nomad LLC.</em></p>
      </div>
    </div>
  </div>
);

export default ReviewListEntry;
