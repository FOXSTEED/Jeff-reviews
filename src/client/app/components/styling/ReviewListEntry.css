.main {
  height: auto;
  display: flex;
  margin-top: 5px;
  overflow: auto;
  margin-bottom: 10px;
  border-bottom-color: rgb(229, 229, 229);
  border-bottom-style: solid;
  border-bottom-width: 1px;
}

.user {
  margin: 0;
  padding: 0;
  height: 40%;
  display: table-row;
  flex-direction: column;
  float: left;
  padding: 12px;
  box-sizing: border-box;
  text-align: center;
}

.belowUser {
  color: rgb(0, 166, 128);
  font-size: 12px;
  margin-left: 2px;
  margin-right: 2px;
  text-align: center;
}

.username {
  font-size: 11px;
  cursor: pointer;
  line-height: 20px;
  color: rgb(118, 118, 118);
  margin: 0;
}

.location {
  font-size: 11px;
  cursor: pointer;
  line-height: 20px;
  color: rgb(44, 44, 44);
  font-weight: 700;
  margin: 0;
}

.picture {
  width: 80px;
  height: 80px;
  border-radius: 100px;
}

.header {
  display: inline-block;
}

.bubbles {
  margin: 5px;
}

.date {
  color: rgb(74, 74, 74);
  font-size: 12px;
}

.headline {
  color: rgb(0, 10, 18);
  font-size: 18px;
  font-style: normal;
  font-weight: 400;
  text-align: left;
}

.review {
  color: rgb(74, 74, 74);
  font-size: 14px;
  line-height: 20px;
}

.ask {
  font-size: 12px;
  color: rgb(0, 102, 153);
  text-decoration: none;
  display: block;
  margin-bottom: 10px;
}

.button {
  background-color: rgb(255, 255, 255);
  color: rgb(74, 74, 74);
  font-size: 12px;
  font-weight: 400;
  height: 20px;
  border: 1px solid #e5e5e5;
  border-radius: 4px;
  display: inline-block;
  cursor: pointer;
}

.flag {
  float: right;
  color: rgb(74, 74, 74);
}

.button:hover {
  border-color: #00a680;
}

.thumb {
  color: #00a680;
  margin-right: 5px;
}

.bottomnote {
  font-size: 12px;
  margin-bottom: 10px;
  border-top-color: rgb(229, 229, 229);
  border-top-width: 1px;
  border-top-style: solid;
  padding-top: 12px;
  color: rgb(118, 118, 118);
  width: 75%;
}