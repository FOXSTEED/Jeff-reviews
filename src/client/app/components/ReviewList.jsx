/* eslint-disable react/prop-types */

import React from 'react';
import ReviewListEntry from './ReviewListEntry';

const ReviewList = props => (
  <div>
    {props.reviews.map(review => <ReviewListEntry review={review} />)}
  </div>
);


export default ReviewList;
