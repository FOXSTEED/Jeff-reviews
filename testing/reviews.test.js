import React from 'react';
import Reviews from '../src/client/app/index';
import ReviewListEntry from '../src/client/app/components/ReviewListEntry';
import ReviewList from '../src/client/app/components/ReviewList';
import data from './dummyData';

describe('Reviews app', () => {
  it('should render Reviews module', () => {
    const wrapper = shallow(<Reviews />);
    expect(wrapper).toMatchSnapshot();
  });
  
  it('should render X amount of reviews', () => {
    const wrapper = shallow(<ReviewList reviews={data.data} />);
    expect(wrapper.find(ReviewListEntry).length).toBe(9);
  });
});
