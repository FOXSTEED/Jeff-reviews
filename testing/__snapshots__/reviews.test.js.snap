// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reviews app should render Reviews module 1`] = `
<div
  className="main"
>
  <div
    className="titleHeader"
  >
    <span
      className="title"
    >
      Reviews
    </span>
    <span
      className="counter"
    >
      (0)
    </span>
    <button
      className="button"
    >
      Write a Review
    </button>
  </div>
  <div
    className="graph"
  >
    <Graph
      graphInfo={Array []}
      handleRating={[Function]}
    />
  </div>
  <div
    className="header"
  >
    <Search
      filter={[Function]}
      numRev={0}
      reset={[Function]}
      words={Array []}
    />
  </div>
  <div>
    <TryAgain
      reset={[Function]}
    />
  </div>
</div>
`;
